codice/ddf
# Prevent CVEs for docs and itests dependencies from appearing in OWASP reports and speed up OWASP scans during CI

#### What does this PR do?
Prevents CVEs for docs and itests dependencies from appearing in OWASP reports
* Changes scope of docs dependencies so they don't leak into distribution
* Adds an "itests" profile so all itest modules can be skipped. Previously we were only skipping the `test-itests-ddf` module so CVEs for the other modules were showing up

Speeds up OWASP scanning during CI
* Run the "aggregate" goal instead of "check".
* Change the "owasp-dist" profile to scan any zip build artifacts (i.e. kernel, distribution) when using the aggregate goal

With these changes, we now get OWASP reports for both PR and branch builds, and the branch builds include CVEs for the distro, kernel, etc. which they didn't before. Also, the scans run much faster. Here's the difference in performance (running on my laptop):
* `aggregate`: 2:24 min
* `aggregate -Powasp-dist`: 6:43 min
* `check -Powasp-dist`: 38:56 min

#### Who is reviewing it? 
@bakejeyner @blen-desta @emmberk @garrettfreibott  @stustison  

#### Select relevant component teams: 
@codice/continuous-integration 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.


#### How should this be tested?
CI for the OWASP scan changes for PRs. Also for the itest stages. Should verify that:
* OWASP report is available
* Report contains no CVEs for docs or itest dependencies
* 'Integration Tests Only Build' stage runs all itests except test-itests-ddf-core

Not sure how to test the OWASP scanning changes for branch builds without merging. 

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #____

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
